xlators/cluster/dht/src/dht-helper.c

```diff
@@ -724,8 +724,7 @@ dht_local_wipe(dht_local_t *local)
     if (local->xattr)
         dict_unref(local->xattr);
 
-    if (local->inode)
-        inode_unref(local->inode);
+    inode_unref(local->inode);
```

Comment:
I don't think it's a good idea to unconditionally call `inode_unref()` (or `dht_layout_unref()` in some other places) assuming that it will already check if the object is NULL and do nothing in that case.

First of all this makes the code inconsistent. There are some objects, like fd's and dict's where we need to explicitly check if the object is NULL or not before calling unref, while there are other cases like layouts and inodes where we don't. This seems arbitrary and it's not good.

From the performance point of view the cost of a comparison and a conditional jump is smaller than a call, so in those cases that the object is NULL, it will be faster (though probably negligibly faster). If not, the code will check the object twice, but this is very cheap.

Reply:
Yes the above seems reasonable. Made the necessary changes.